### PR TITLE
Extend contribution guidelines (new commands, services, verbosity)

### PR DESCRIPTION
**Pull request by joaopalet**

### CODE REVIEW

File: CONTRIBUTION.md [hunk 1]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+
+### Implementing a new command
+
+Let's suppose you want to want to implement a new command `bar`, that would be the direct child of an existing command `stackit foo` (meaning it would be invoked as `stackit foo bar`):
+
+1. You would start by creating a new folder `bar/` inside `internal/cmd/foo/`
+2. Following with the creation of a file `bar.go` inside your new folder `internal/cmd/foo/bar/`
+   1. The Go package should be similar to the command usage, in this case `package bar` would be an adequate name
+   2. Please refer to the [Command file structure](./CONTRIBUTION.md/#command-file-structure) section for details on the strcutre of the file itself
+3. To register the command `bar` as a child of the exisitng command `foo`, add `cmd.AddCommand(bar.NewCmd(p))` to the `addSubcommands` method of the constructor of the `foo` command
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:49]:
```suggestion
3. To register the command `bar` as a child of the existing command `foo`, add `cmd.AddCommand(bar.NewCmd(p))` to the `addSubcommands` method of the constructor of the `foo` command
```

File: CONTRIBUTION.md [hunk 2]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+   1. In this case, `p` in this case is the `printer` that is passed from the root command to all subcommands of the tree (refer to the [Outputs, prints and debug logs](./CONTRIBUTION.md/#outputs-prints-and-debug-logs) section for more details regarding the `printer`)
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:50]:
```suggestion
   1. In this case, `p` is the `printer` that is passed from the root command to all subcommands of the tree (refer to the [Outputs, prints and debug logs](./CONTRIBUTION.md/#outputs-prints-and-debug-logs) section for more details regarding the `printer`)
```

File: CONTRIBUTION.md [hunk 3]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+package bar
+
+import (
+	(...)
+)
+
+// Define consts for command flags
+const (
+   someArg = "MY_ARG"
+	someFlag = "my-flag"
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:68]: wrong indentation?

File: CONTRIBUTION.md [hunk 4]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+	MyFlag *string
+}
+
+// "bar" command constructor
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "bar",
+		Short: "Short description of the command (is shown in the help of parent command)",
+		Long:  "Long description of the command. Can contain some more information about the command usage. It is shown in the help of the current command.",
+		Args:  args.SingleArg(instanceIdArg, (utils.ValidateUUID)), // Validate argument
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:85]:
```suggestion
		Args:  args.SingleArg(someArg, utils.ValidateUUID), // Validate argument, with an optional validation function
```

File: CONTRIBUTION.md [hunk 5]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+			if err != nil {
+				return fmt.Errorf("(...): %w", err)
+			}
+
+         projectLabel, err := projectname.GetProjectName(ctx, cmd)
+				if err != nil {
+					projectLabel = model.ProjectId
+				}
+
+         // Check API response "resp" an output accordingly
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:120]:
```suggestion
         // Check API response "resp" and output accordingly
```

File: CONTRIBUTION.md [hunk 6]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+	return &inputModel{
+		GlobalFlagModel: globalFlags,
+      MyArg            myArg,
+		MyFlag:          flags.FlagToStringPointer(cmd, myFlag),
+	}, nil
+}
+
+// Build request to the API
+func buildRequest(ctx context.Context, model *inputModel, apiClient *foo.APIClient) (foo.ApiListInstancesRequest error) {
+   someParam, err := utils.GetParam(model.MyFlag) // Util function to get API parameter based on flag value
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:156]: this is not the most common case, so for simplicity I would remove it

File: CONTRIBUTION.md [hunk 7]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+   }
+	req := apiClient.GetBar(ctx, model.ProjectId, model.MyArg, someParam)
+	return req
+}
+
+// Output result based on the configured output format
+func outputResult(cmd *cobra.Command, outputFormat string, resources []foo.Resource) error {
+	switch outputFormat {
+	case globalflags.JSONOutputFormat:
+		details, err := json.MarshalIndent(instances, "", "  ")
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:168]:
```suggestion
		details, err := json.MarshalIndent(resources, "", "  ")
```

File: CONTRIBUTION.md [hunk 8]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+The CLI has 4 different verbosity levels:
+
+- `error`: For only displaying errors
+- `warning`: For displaying user facing warnings _(and all of the above)_
+- `info` (default): For displaying user facing info, such as operation success messages and spinners _(and all of the above)_
+- `debug`: For displaying structured logs with different levels, including errors _(and all of the above)_
+
+For prints that are specific to a certain log level, you can use the methods defined in the `print` package: `Error`, `Warn`, `Info`, and `Debug`.
+
+For command outputs that should always be displayed, no matter the defined verbosity, you should use the `print` methods `Outputf` and `Outputln`.
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:205]: Make it more clear that these methods should only be used for the command output, which are the outputs described by "I run the command to see this"

File: CONTRIBUTION.md [hunk 9]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+           apiClient, err = foo.NewAPIClient(cfgOptions...)
+           if err != nil {
+              return nil, &errors.AuthError{}
+           }
+
+           return apiClient, nil
+        }
+        ```
+
+2.  Add a `FooCustomEndpointKey` key in `internal/pkg/config/config.go` (and add it to `ConfigKeys` and set the to default to `""` using `viper.SetDefault`)
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:250]: Steps 2 and 3 should come in the beginning. Otherwise in the client we are using consts that are not yet defined

Comment (vicentepinto98) [inline CONTRIBUTION.md:250]: add a small snippet here maybe?

File: CONTRIBUTION.md [hunk 10]
```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
...
+3.  Update the `stackit config unset` and `stackit config unset` commands by adding flags to set and unset a custom endpoint for the `foo` service API, respectively, and update their unit tests
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:251]: also here

File: CONTRIBUTION.md [hunk 11]
```diff
@@ -4,9 +4,18 @@ Your contribution is welcome! Thank you for your interest in contributing to the
 
 ## Table of contents
 
-- [Developer Guide](#developer-guide)
-- [Code Contributions](#code-contributions)
-- [Bug Reports](#bug-reports)
+- [Contribute to the STACKIT CLI](#contribute-to-the-stackit-cli)
+  - [Table of contents](#table-of-contents)
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:8]: Nitpick, but is it common to have the table of contents in the table of contents? :sweat_smile: 
I think we can start with `Developer Guide` at the top

File: CONTRIBUTION.md (hunk 9 above)
Comment (joaopalet) [reply CONTRIBUTION.md:250]: I think having more snippets might make the document more complex and harder to read, in this case it becomes clear once the user opens the file and sees all other uses for other services

File: CONTRIBUTION.md (hunk 10 above)
Comment (joaopalet) [reply CONTRIBUTION.md:251]: Same as above

File: CONTRIBUTION.md (hunk 11 above)
Comment (joaopalet) [reply CONTRIBUTION.md]: Hm, I'm not sure why but the table of contents is automatically updated every time I save the file, and it includes _all_ headings. I've removed it and saved without formatting

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**